Add test  webFilter in Soul-web

- [ ]  CrossFilter  @Trafalgar-YuI
- [ ]  FileSizeFilter  @hellboy0621 
- [ ]  WebSocketParamFilter  @marina432

> get FileSizeFilter

When I run the code bellow, the error occurs.
java.lang.AssertionError: expectation "expectComplete" failed (expected: onComplete(); actual: onError(java.lang.NullPointerException: The Mono returned by the supplier is null))

I need help.

`ServerWebExchange webExchange =
		MockServerWebExchange.from(MockServerHttpRequest
				.post("http://localhost:8080")
				.contentType(MediaType.TEXT_PLAIN)
				.contentLength(1)
				.body("a"));

ServerHttpRequest mutatedRequest = webExchange.getRequest().mutate().header(CONTENT_TYPE,
		String.valueOf(MULTIPART_FORM_DATA)).build();
webExchange = webExchange.mutate().request(mutatedRequest).build();

mutatedRequest = webExchange.getRequest().mutate().header(CONTENT_LENGTH,
		String.valueOf(10000)).build();
webExchange = webExchange.mutate().request(mutatedRequest).build();


WebFilterChain webFilterChain = mock(WebFilterChain.class);
when(webFilterChain.filter(webExchange)).thenReturn(Mono.empty());

FileSizeFilter fileSizeFilter = new FileSizeFilter(1);
Mono<Void> voidMono = fileSizeFilter.filter(webExchange, webFilterChain);
StepVerifier.create(voidMono).expectSubscription().verifyComplete();`

--- a/soul-admin/src/main/java/org/dromara/soul/admin/listener/http/HttpLongPollingDataChangedListener.java
+++ b/soul-admin/src/main/java/org/dromara/soul/admin/listener/http/HttpLongPollingDataChangedListener.java
@@ -176,7 +176,7 @@ public class HttpLongPollingDataChangedListener extends AbstractDataChangedListe
     }
 
     private List<ConfigGroupEnum> compareChangedGroup(final HttpServletRequest request) {
-        List<ConfigGroupEnum> changedGroup = new ArrayList<>(4);
+        List<ConfigGroupEnum> changedGroup = new ArrayList<>(ConfigGroupEnum.values().length);
         for (ConfigGroupEnum group : ConfigGroupEnum.values()) {
             // md5,lastModifyTime
             String[] params = StringUtils.split(request.getParameter(group.name()), ',');

--- a/soul-web/src/test/java/org/dromara/soul/web/filter/FileSizeFilterTest.java
+++ b/soul-web/src/test/java/org/dromara/soul/web/filter/FileSizeFilterTest.java
@@ -1,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dromara.soul.web.filter;
+
+import org.dromara.soul.plugin.api.result.SoulResult;
+import org.dromara.soul.plugin.base.utils.SpringBeanUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.http.MediaType;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.mock.http.server.reactive.MockServerHttpRequest;
+import org.springframework.mock.web.server.MockServerWebExchange;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilterChain;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static org.springframework.http.MediaType.MULTIPART_FORM_DATA;
+
+/**
+ * test case for FileSizeFilter.
+ *
+ * @author hellboy0621
+ **/
+public class FileSizeFilterTest {
+
+    @Before
+    public void setup() {
+        GenericApplicationContext context = new GenericApplicationContext();
+        SpringBeanUtils.getInstance().setCfgContext(context);
+        context.getBeanFactory().registerSingleton("soulResult", mock(SoulResult.class));
+        context.refresh();
+    }
+
+    @Test
+    public void testFilter() {
+        ServerWebExchange webExchange =
+                MockServerWebExchange.from(MockServerHttpRequest
+                        .post("http://localhost:8080")
+                        .contentType(MediaType.TEXT_PLAIN)
+                        .contentLength(4)
+                        .body("test"));
+
+        ServerHttpRequest mutatedRequest = webExchange.getRequest().mutate().header(CONTENT_TYPE,
+                String.valueOf(MULTIPART_FORM_DATA)).build();
+        webExchange = webExchange.mutate().request(mutatedRequest).build();
+
+        WebFilterChain webFilterChain = mock(WebFilterChain.class);
+        when(webFilterChain.filter(any())).thenReturn(Mono.empty());
+
+        FileSizeFilter fileSizeFilter = new FileSizeFilter(0);
+        Mono<Void> voidMono = fileSizeFilter.filter(webExchange, webFilterChain);
+        StepVerifier.create(voidMono).expectSubscription().verifyComplete();
+
+        fileSizeFilter = new FileSizeFilter(1);
+        voidMono = fileSizeFilter.filter(webExchange, webFilterChain);
+        StepVerifier.create(voidMono).expectSubscription().verifyComplete();
+    }
+}

